rust-only-stdlib/src/main.rs

```diff
@@ -0,0 +1,81 @@
+use std::io;
+use std::io::prelude::Write;
+use std::io::{BufRead, BufReader};
+use std::net::{TcpListener, TcpStream};
+use std::time::SystemTime;
+
+extern "C" {
+    fn ctime(timep: *const u64) -> *const u8;
+    fn strlen(s: *const u8) -> usize;
+}
+
+fn current_timestamp() -> String {
+    unsafe {
+        let timestamp_in_seconds = match SystemTime::now().duration_since(SystemTime::UNIX_EPOCH) {
+            Ok(duration) => duration.as_secs(),
+            Err(e) => panic!("Failed to calculate duration: {}", e),
+        };
+        let timestamp = ctime(&timestamp_in_seconds);
+        String::from_utf8_unchecked(
+            std::slice::from_raw_parts(timestamp, strlen(timestamp) - 1).to_vec(),
+        )
+        .trim()
+        .to_string()
+    }
+}
+
+fn build_response(stream: &TcpStream) -> io::Result<String> {
+    let mut req = String::with_capacity(512);
+    let mut reader = BufReader::new(stream);
+    reader.read_line(&mut req)?;
+
+    println!("Request:\n{}", req);
+    let name = match req.split('?').nth(1) {
+        Some(q) => q
+            .split('=')
+            .nth(1)
+            .expect("No '=' after '?'")
+            .split_whitespace()
+            .nth(0)
+            .unwrap(),
+        None => "World",
+    };
+    let response_body = format!("{{\"message\": \"Hello {}!\"}}", name);
+    let content_length = response_body.len();
```

Comment:
I get

```
< HTTP/1.1 200 OK
< Content-Type: application/json
< Date: Mon Mar 25 08:29:22 2019
< Content-Length: 25
<
* Excess found in a non pipelined read: excess = 1, size = 25, maxdownload = 25, bytecount = 0
* Connection #0 to host localhost left intact
```

Looks to me like the actual content length is longer than `response_body.len()` by a newline in the end.

Reply:
Where do you get this error message? Everything looked good using curl and Firefox.
I removed the newline after my response body anyway, maybe this will fix it

Reply:
That fixes it!
I just curled `curl -X POST -v "localhost:8080/say-hello?name=Eva"`. The warning appears if you enable verbose logging. And I saw in the shell that it was outputting a newline at the end :man_shrugging:

Reply:
ah, I was missing the verbose logging :)